src/triage/component/catwalk/predictors.py

```diff
@@ -342,28 +260,54 @@ def predict(self, model_id, matrix_store, misc_db_parameters, train_matrix_colum
                 session.close()
 
         model = self.load_model(model_id)
-        logger.spam(f"Loaded model {model_id}")
         if not model:
-            raise ModelNotFoundError(f"Model # {model_id} not found")
+            raise ValueError(f"Model id {model_id} not found")
+        logger.spam(f"Loaded model {model_id}")
 
         # Labels are popped from matrix (i.e. they are removed and returned)
         labels = matrix_store.labels
 
-        predictions_proba = model.predict_proba(
+        predictions = model.predict_proba(
             matrix_store.matrix_with_sorted_columns(train_matrix_columns)
-        )
+        )[:, 1]  # Returning only the scores for the label == 1
+
+
         logger.debug(
             f"Generated predictions for model {model_id} on {matrix_store.matrix_type.string_name} matrix {matrix_store.uuid}"
         )
         if self.save_predictions:
+            df = pd.DataFrame(data=None, columns=None, index=matrix_store.index)
+            df['label_value'] = matrix_store.labels
+            df['score'] = predictions
+
+
+            logger.spam(f"Sorting predictions for model {model_id} using {self.rank_order}")
+
+            if self.rank_order == 'best':
+                df.sort_values(by=["score", "label_value"], inplace=True, ascending=[False,False], na_position='last')
+            elif self.rank_order == 'worst':
+                df.sort_values(by=["score", "label_value"], inplace=True, ascending=[False,True], na_position='first')
+            elif self.rank_order == 'random':
+                df.sample(frac=1).sort_values(by='score', inplace=True, ascending=False)
```

Comment:
does `df.sample()` operate in place? I would have expected this to return a new object?

Reply:
You are right! Good eye!

Fixing it

Reply:
Hmmm... this looks like it's gone back to what it was before, but the `sample` call is still going to return a new df, so I don't think this will yield the intended result of shuffling the dataframe. Pretty sure this needs to be:
```
df = df.sample(frac=1).sort_values(by='score', inplace=False, ascending=False)
```

For the sort, `inplace` needs to be `False` to return the result as a new dataframe as well.